### DISCUSSION TITLE
best way to capitalize a string in js, use a function

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [elchininet/custom-sidebar#219] [Q&A] [answered]
for several info texts I use a template that returns a lowercase string

example:
```
js_variables:
# data
  trash: sensor.afvalwijzer_vandaag

partials:
  data: |
    const afval = states(trash)
```

but I want it to be displayed capitalized. Jinja has that function and filter `|capitalize`, javascript requires us to create it.

with `custom:button-card` I use the string slice method and set the first toUpperCase() and attach the remaining. in sidebar syntax:

```
  data: |
    const afval = states(trash).charAt(0).toUpperCase() + states(trash).slice(1);
```

would you think this is the best way to go, or could we maybe install the function first as some generic method in the config, to be able to call it in other templates too?

not sure where we could do that, but in it would be something like:
```
function capitalizeFirstLetter(string) {
        return string.charAt(0).toUpperCase() + string.slice(1);
        }
```

and then use it  with:
```
[[[ return capitalizeFirstLetter(afval); ]]]
```

given this would be used throughout the config and not in a singled out dashboard, I can not use it as partial.

or, would be even better.... have the function built in into the plugin ?

**Comment by elchininet** [comment, marked as answer]
>would you think this is the best way to go

Yes, you are already doing it in the right way. There are many ways in which you can write such a function, this is just one of them (remember that you can include a partial inside another partial):

```yaml
js_variables:
  trash: sensor.afvalwijzer_vandaag
  temperature: sensor.temperature
partials:
  utilities: |
    const capitalize = str => str[0].toUpperCase() + str.slice(1);
  data: |
    @partial utilities
    const afval = capitalize(states(trash));
order:
  - item: something
    name: |
      [[[
        @partial data
        return afval;
      ]]]
    info: |
      [[[
        @partial utilities
        return capitalize(states(temperature));
      ]]]
```

**Reply by Mariusthvdb** [reply]
thanks! that will do just fine.
magic
